libraries/heft-config-file/src/ConfigurationFileBase.ts

```diff
@@ -16,6 +16,8 @@ interface IConfigurationJson {
  * @beta
  *
  * The set of possible mechanisms for merging properties from parent configuration files.
+ * If a child configuration file sets a property value to `null`, that will always delete the value
+ * specified in the parent configuration file, regardless of the inheritance type.
```

Comment:
> This follows prior art in TypeScripts tsconfig inheritance, which uses the same method to reset a value to "not specified".

Generally I prefer to avoid `null` everywhere, because it causes confusion with `undefined`.  However this convention of using `null` to reset a property seems fine, because the final parsed object will not contain `null`. 👍

Reply:
This really comes down to that `null` is allowed in JSON, but `undefined` isn't, so it's the value of choice.